### PR TITLE
handle_lost_connection

### PR DESCRIPTION
**Pull request by dirvine**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/596)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/routing_node.rs [hunk 1]
```diff
@@ -167,18 +167,19 @@ impl RoutingNode {
...
-            }
-        }
+        // FIXME(dirvine) I am not sure we should not accept connections here :16/08/2015
+        // let has_bootstrap_endpoints = self.core.has_bootstrap_endpoints();
+        // if !self.core.is_node() {
+        //     if has_bootstrap_endpoints {
+        //         // we are bootstrapping, refuse all normal connections
+        //         self.connection_manager.drop_node(endpoint);
+        //         return;
+        //     } else {
```
Comment (benjaminbollen) [inline src/routing_node.rs]: We can debate whether "normal connections" should not be accepted, but by disabling the self_assignment of a name the network will for sure not start, as everyone will stay a Client, and no-one will become a Node.

The condition upon which to refuse connections here is very subtle, and tests out to be correct - at least in the current implementation:
 if ! core.is_node()  ===>   if we do not have a Node(name)/ a routing table
    if has_bootstrap_endpoints ===> but we do have connections that are labelled as bootstrap;
===> we are connected to relay nodes (either trying to request a network name, or as pure clients), so we should not attempt to help other nodes bootstrap into the network, as we are ourselves still clients

Comment (dirvine) [inline src/routing_node.rs]: Updated now

File: src/routing_node.rs [hunk 2]
```diff
@@ -190,8 +191,9 @@ impl RoutingNode {
 
     /// When CRUST reports a lost connection, ensure we remove the endpoint anywhere
     fn handle_lost_connection(&mut self, endpoint : Endpoint) {
-        error!("Lost connection on {:?}, but CORE IS NOT UPDATED!", endpoint);
-        //unimplemented!()
+        error!("Lost connection on {:?}", endpoint);
+        let connection_name = self.core.lookup_endpoint(&endpoint);
```
Comment (benjaminbollen) [inline src/routing_node.rs]: It is possibly more desirable to have an explicit function core::drop_endpoint (currently only have drop_peer), but that's up for discussion

Comment (dirvine) [inline src/routing_node.rs]: Yes agreed, it will do for now I suppose. Not to bad.

**Comment by benjaminbollen** [comment]
looks good to me !

**Pull request merged**